app.rb

```diff
@@ -0,0 +1,106 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'rental'
+# require_relative 'classroom'
+require_relative 'teacher'
+# require 'pry'
+
+class App
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  # List all books.
+  def list_books
+    puts 'List of Books:'
+    @books.each do |book|
+      puts "📚 Title: #{book.title} by #{book.author}"
+    end
+  end
+
+  # List all people.
+  def list_people
+    @persons.each do |person|
+      if person.instance_of?(Student)
+        puts "[Student 🧍], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      else
+        puts "[Teacher 👨‍🏫], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      end
+    end
+  end
+
+  # Create a person (teacher or student)
+  def create_a_person
+    puts 'Do you want to add a student (1) or a teacher (2)? [Insert the number]'
+    is_student = gets.chomp.to_i
+    puts 'Age'
+    age = gets.chomp.to_i
+    puts 'Name'
+    person_name = gets.chomp
+
+    case is_student
+    when 1
+      puts 'Has parent permission? [Y/N]'
+      gets.chomp.downcase
+      student = Student.new(age: age, name: person_name)
```

Comment:
- When trying to create a teacher, I get this error. Check your student class in your initialize to see how you've added the parameters to fix this.

![image](https://github.com/ANTOINE1128/OOP--School--Library/assets/59397159/43812995-1f0d-4a64-94a4-b69120511ea7)